python/tests/core/test_statistics_engine.py

```diff
@@ -360,7 +360,77 @@ def test_compute_statistics_feature_view_obj_feature_group_commit_id(self, mocke
         assert mock_statistics_engine_save_statistics.call_count == 0
         assert mock_engine_get_instance.return_value.profile_by_spark.call_count == 0
 
-    def test_profile_statistics(self, mocker):
+    def test_compute_single_feature_statistics(self, mocker):
+        # Arrange
+        feature_store_id = 99
+
+        mocker.patch("hsfs.client.get_instance")
+        mocker.patch("hsfs.engine.get_type")
+        mock_statistics_engine_profile_statistics = mocker.patch(
+            "hsfs.core.statistics_engine.StatisticsEngine.profile_statistics",
+            return_value=None,
+        )
+        mock_statistics_engine_save_statistics = mocker.patch(
+            "hsfs.core.statistics_engine.StatisticsEngine._save_statistics"
+        )
+        mock_engine_get_instance = mocker.patch("hsfs.engine.get_instance")
+
+        s_engine = statistics_engine.StatisticsEngine(feature_store_id, "featuregroup")
+
+        # Act
+        with pytest.raises(exceptions.FeatureStoreException) as e_info:
+            s_engine.compute_single_feature_statistics(
+                feature_dataframe=None,
+                feature_name=None,
+            )
+
+        # Assert
+        assert mock_statistics_engine_profile_statistics.call_count == 0
+        assert mock_statistics_engine_save_statistics.call_count == 0
+        assert mock_engine_get_instance.return_value.profile_by_spark.call_count == 0
+        assert (
+            str(e_info.value)
+            == "Descriptive statistics for feature monitoring cannot be computed from the Python engine."
+        )
+
+    def test_compute_single_feature_statistics_get_type_spark(self, mocker):
+        # Arrange
+        feature_store_id = 99
+
+        mocker.patch("hsfs.client.get_instance")
+        mocker.patch("hsfs.engine.get_type", return_value="spark")
+        mocker.patch("hsfs.util.get_hudi_datestr_from_timestamp")
+        mock_statistics_engine_profile_statistics = mocker.patch(
+            "hsfs.core.statistics_engine.StatisticsEngine.profile_statistics",
+            return_value="",
+        )
+        mocker.patch("hsfs.statistics.Statistics")
+        mocker.patch("hsfs.feature_group.FeatureGroup.select_all")
+        mocker.patch("hsfs.feature_group.FeatureGroup.read")
```

Comment:
That is a lot of mocking, can we not get a cleaner version with minimal mocking?